src/detect-engine-port.c

```diff
@@ -2466,6 +2471,85 @@ static int PortTestMatchDoubleNegation(void)
     return result;
 }
 
+
+// Test that negation is successfully parsed with whitespace for port strings of
+// length < 16
+static int DetectPortParseDoTest(void)
+{
+    int result = 0;
+    DetectEngineCtx *de_ctx = DetectEngineCtxInit();
+    DetectPort *head = NULL;
+    DetectPort *nhead = NULL;
+    const char *str = "[30:50, !45]";
+    int r = DetectPortParseDo(de_ctx, &head, &nhead, str, 0, NULL, 0);
+    
+    // Assertions
+    FAIL_IF_NULL(head);
+    FAIL_IF_NULL(nhead);
+    FAIL_IF(r < 0);
+    FAIL_IF(head->port != 30);
+    FAIL_IF(head->port2 != 50);
+    FAIL_IF(nhead->port != 45);
+    FAIL_IF(nhead->port2 != 45);
+    PASS;
+}
+
+// Tests that port strings of length == 16 fail to parse
+static int DetectPortParseDoTest2(void)
+{
+    DetectEngineCtx *de_ctx = DetectEngineCtxInit();
+    DetectPort *head = NULL;
+    DetectPort *nhead = NULL;
+    const char *str = "[30:50,              !45]";
+    int r = DetectPortParseDo(de_ctx, &head, &nhead, str, 0, NULL, 0);
+    FAIL_IF(r >= 0);
+    PASS;
+}
+
+
+// Verifies correct parsing when negation port string length < 16
+static int PortParseTestLessThan14Spaces(void) 
+{
+    const char *str = "       45";
+    DetectPort *dp = NULL;
+    dp = PortParse(str);
+
+    FAIL_IF(dp->port != 45);
+    FAIL_IF(dp->port2 != 45);
+    PASS;
+}
+
+// Verifies NULL returned when negation port string length == 16
+static int PortParseTest14Spaces(void) 
+{
+    const char *str = "              45";
+    DetectPort *dp = NULL;
+    dp = PortParse(str);
+    FAIL_IF_NOT_NULL(dp);
+    PASS;
+}
+
+// Verifies NULL returned when negation port string length >= 16
+static int PortParseTestMoreThan14Spaces(void) 
+{
+    const char *str = "                                   45";
+    DetectPort *dp = NULL;
+    dp = PortParse(str);
+    FAIL_IF_NOT_NULL(dp);
+    PASS;
+}
+
+// DetectPortParseTest
+static int DetectPortParseTest(void)
+{
+    DetectEngineCtx *de_ctx = DetectEngineCtxInit();
+    DetectPort *head = NULL;
+    const char *sigstr = "[30:50,              !45]";
+    DetectPortParse(de_ctx, &head, sigstr);
+    return 1;
```

Comment:
what is tested here?

Reply:
oops. I think I wrote that initially as an entrypoint for my debugger. I'll remove it